paymentsheet/src/main/java/com/stripe/android/lpmfoundations/paymentmethod/PaymentMethodMetadata.kt

```diff
@@ -58,13 +58,39 @@ internal data class PaymentMethodMetadata(
             } else {
                 paymentMethodDefinition.supportedPaymentMethod(this, sharedDataSpec)
             }
-        }.paymentMethodSorter()
+        }.run {
+            // Optimization to early out if we don't have a client side order.
+            if (paymentMethodOrder.isNotEmpty()) {
+                val orderedPaymentMethodTypes = orderedPaymentMethodTypes().mapOrderToIndex()
+                sortedBy { supportedPaymentMethod ->
```

Comment:
highly likely I'm missing something here, but why is this necessary? I would've thought that orderedPaymentMethodTypes() returns the payment methods already sorted

Reply:
got it, thanks for the explanation!!